task3_snkrs/spiders/lanebryant_spider.py

```diff
@@ -0,0 +1,116 @@
+import json
+
+from scrapy.linkextractors import LinkExtractor
+from scrapy.spiders import CrawlSpider, Rule
+from w3lib.url import add_or_replace_parameters
+
+from task3_snkrs.items import LanebryantItem
+
+
+class LanebryantSpider(CrawlSpider):
+    name = "lanebryant"
+
+    allowed_domains = [
+        'lanebryant.com',
+        'lanebryant.scene7.com',
+    ]
+    start_urls = [
+        'https://www.lanebryant.com',
+    ]
+    listings_css = [
+        '.mar-subnav-links-column',
+        '.mar-pagination a',
+    ]
+    rules = [
+        Rule(LinkExtractor(restrict_css=listings_css)),
+        Rule(LinkExtractor(restrict_css='.mar-prd-item-image-container'), callback='parse_product'),
+    ]
+
+    def parse_product(self, response):
+        data = json.loads(response.css('#pdpInitialData::text').getall()[0])
```

Comment:
Let's create a separate method for raw_product. As we do not do any computation in `parse` method.